### PR TITLE
Added hitgroup to CTakeDamageInfo

### PR DESCRIPTION
**Pull request by schwarper**
image

```csharp
public static HookResult OnTakeDamage(DynamicHook hook)
{
    CEntityInstance entity = hook.GetParam<CEntityInstance>(0);

    if (entity.DesignerName != "player")
    {
        return HookResult.Continue;
    }

    var info = hook.GetParam<CTakeDamageInfo>(1);

    Server.PrintToChatAll($"HEY {info.Attacker.Value?.DesignerName} -> {info.GetHitGroup()}");
    return HookResult.Continue;
}
```

### CODE REVIEW

**Comment by roflmuffin** [comment]
The changed file is a generated file so you'll need to make a new file with a partial class to extend the generated code

File: managed/CounterStrikeSharp.API/Core/Schema/Classes/CTakeDamageInfo.g.cs [hunk 1]
```diff
@@ -101,5 +101,4 @@ public CTakeDamageInfo (IntPtr pointer) : base(pointer) {}
 	// m_bInTakeDamageFlow
 	[SchemaMember("CTakeDamageInfo", "m_bInTakeDamageFlow")]
 	public ref bool InTakeDamageFlow => ref Schema.GetRef<bool>(this.Handle, "CTakeDamageInfo", "m_bInTakeDamageFlow");
-
-}
+}
```
Comment (roflmuffin) [inline managed/CounterStrikeSharp.API/Core/Schema/Classes/CTakeDamageInfo.g.cs:104]: Can you please reset this file so its no longer in the diff?

File: managed/CounterStrikeSharp.API/Core/Model/CTakeDamageInfo.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        nint v1 = *(nint*)(v4 + 16);
+
+        if (v1 == nint.Zero)
+        {
+            return HitGroup_t.HITGROUP_GENERIC;
+        }
+
+        return (HitGroup_t)(*(uint*)(v1 + 56));
+    }
```
Comment (roflmuffin) [inline managed/CounterStrikeSharp.API/Core/Model/CTakeDamageInfo.cs:29]: You should be able to convert this to safe code (untested):

```suggestion
    public HitGroup_t HitGroup()
    {
        IntPtr v4 = Marshal.ReadIntPtr(handle, 0x78);

        if (v4 == nint.Zero)
        {
            return HitGroup_t.HITGROUP_INVALID;
        }

        IntPtr v1 = Marshal.ReadIntPtr(v4, 16

        if (v1 == nint.Zero)
        {
            return HitGroup_t.HITGROUP_GENERIC;
        }

        return (HitGroup_t)Marshal.ReadInt32(v1, 56);
    }
```

Comment (schwarper) [reply managed/CounterStrikeSharp.API/Core/Model/CTakeDamageInfo.cs]: @roflmuffin yes it's working.

**Comment by KillStr3aK** [comment]
this also could be a property instead? (by name)

or could be 'GetHitGroup'?

**Comment by schwarper** [comment]
> this also could be a property instead? (by name)
> 
> or could be renamed* 'GetHitGroup'?

I guess GetHitGroup is better than HitGroup.

**Comment by KillStr3aK** [comment]
> > this also could be a property instead? (by name)
> 
> > 
> 
> > or could be renamed* 'GetHitGroup'?
> 
> 
> 
> I guess GetHitGroup is better than HitGroup.

indeed, but 'HitGroyp' also stands good if its a property, not a method

**Review by roflmuffin — APPROVED** [review]

**Pull request merged**